ci/test-sanity.sh

```diff
@@ -70,4 +71,16 @@ EOF
 )
 fi
 
+# Disallow (re)introduction of solana sdk dependencies
+(
+  if git diff "$target" | grep -v '+++' | grep '^+.*solana[-_]sdk[: =]'; then
```

Comment:
pretty sure this will be safe. the remaining solana sdk instances are single-line, so shouldn't reasonably need edited before we extricate them

Reply:
could we merge https://github.com/anza-xyz/agave/pull/6235 before this change 🙈 or maybe we can only check lock files 🤔

Reply:
well this also prevents adding new dependencies where solana-sdk hasn't yet been fully removed, which lockfile only checks wouldn't get.

the stuff in #6235 will go away as we completely remove solana-sdk